src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointOptions.tsx

```diff
@@ -1,23 +1,42 @@
-import React from "react";
+import React, { Suspense, useContext, useMemo } from "react";
 
+import Loader from "replay-next/components/Loader";
 import SyntaxHighlightedLine from "replay-next/components/sources/SyntaxHighlightedLine";
+import { ReplayClientContext } from "shared/client/ReplayClientContext";
 import { Point } from "shared/client/types";
-import { getSelectedSource, getSourceContent } from "ui/reducers/sources";
-import { useAppSelector } from "ui/setup/hooks";
-
-import { getTextAtPosition } from "../../../utils/source";
+import { getSourceLinesSuspense } from "ui/suspense/sourceCaches";
 
 type BreakpointProps = {
   type: "breakpoint" | "logpoint";
   breakpoint: Point;
 };
 
-export default function BreakpointOptions({ breakpoint, type }: BreakpointProps) {
-  const selectedSource = useAppSelector(getSelectedSource);
-  const sourceContent = useAppSelector(state =>
-    selectedSource ? getSourceContent(state, selectedSource.id) : null
-  );
+function BreakpointLineContents({ breakpoint }: BreakpointProps) {
+  const replayClient = useContext(ReplayClientContext);
+  const { sourceId } = breakpoint.location;
+
+  const sourceLines = getSourceLinesSuspense(replayClient, sourceId);
+
+  const snippet = useMemo(() => {
+    const { column, line = 0 } = breakpoint.location;
+    let snippet = "";
 
+    if (sourceLines.length > 0) {
+      const lineText = sourceLines[line - 1];
+      return lineText.slice(column, column! + 100).trim();
+    }
+
+    return snippet;
+  }, [sourceLines, breakpoint]);
+
+  if (!snippet) {
+    return null;
+  }
+
+  return <SyntaxHighlightedLine code={snippet} />;
```

Comment:
I wonder if we should read from the syntax parsing cache rather than the source cache? Sometimes parsing a small snippet in isolation will yield bad results.

Reply:
@markerikson Looks like this already got merged, but it seems like this would be worth doing.

Reply:
Yeah, filed FE-1208 for that.

Reply:
Oh! Thanks I hadn't even noticed that.

Reply:
I only filed it after you made that comment :)